src/main/java/fr/insee/bpm/metadata/reader/ReaderUtils.java

```diff
@@ -0,0 +1,84 @@
+package fr.insee.bpm.metadata.reader;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import fr.insee.bpm.exceptions.MetadataParserException;
+import fr.insee.bpm.metadata.Constants;
+import fr.insee.bpm.metadata.model.*;
+import fr.insee.bpm.metadata.reader.lunatic.LunaticReader;
+import fr.insee.bpm.utils.json.JsonReader;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+import static fr.insee.bpm.metadata.reader.ddi.DDIReader.getMetadataFromDDI;
+
+@UtilityClass
+@Slf4j
+public class ReaderUtils {
+
+    public static void addLunaticVariable(MetadataModel metadataModel, String varName, String prefixOrSuffix, VariableType varType) {
+        String correspondingVariableName = varName.replace(prefixOrSuffix, "");
+        Group group;
+        if (metadataModel.getVariables().hasVariable(correspondingVariableName)) { // the variable is directly found
+            group = metadataModel.getVariables().getVariable(correspondingVariableName).getGroup();
+        } else if (metadataModel.getVariables().isInQuestionGrid(correspondingVariableName)) { // otherwise, it should be from a question grid
+            group = metadataModel.getVariables().getQuestionGridGroup(correspondingVariableName);
+        } else {
+            group = metadataModel.getGroup(metadataModel.getGroupNames().getFirst());
+            log.warn(String.format(
```

Comment:
for logging avoid format, use log.warn( {}, var) instead.
And concat the two logs